src/services/hsm.service.js

```diff
@@ -0,0 +1,215 @@
+  delete returnObject.reason;
+  returnObject.isValidXpriv = true;
+  return returnObject;
+}
+
+/**
+ * Derivates an HTR xPriv key from a given xPriv key
+ * @param {Object} hsmConnection
+ * @param {string} hsmKeyName
+ * @param {Object} [options]
+ * @param {boolean} [options.isReadOnlyWallet] If true, derivation will stop at the account level
+ * @param {number} [options.version] Optional, advanced. Version to calculate the exported xPriv
+ * @returns {Promise<{success: boolean, htrKeyName: string}>}
+ */
+async function derivateHtrCkd(
+  hsmConnection,
+  hsmKeyName,
+  options,
+) {
+  const childKeyNames = {
+    HTR_CKD_BIP_KEYNAME: 'HTR_BIP_KEY',
+    HTR_CKD_COIN_KEYNAME: 'HTR_COIN_KEY',
+    HTR_CKD_ACCOUNT_KEYNAME: 'HTR_ACCOUNT_KEY',
+    HTR_CKD_CHANGE_KEYNAME: 'HTR_CHANGE_KEY',
+    HTR_CKD_ADDRESS_KEYNAME: 'HTR_ADDRESS_KEY',
+  };
+
+  /*
+   * Derivation will be made on
+   * BIP code / Coin / Account /  Change
+   *    m/44' / 280' /      0' /       0
+   *        1 /    2 /      3  /       4
+   */
+
+  // Defining derivation version
+  let derivationVersion = options?.version;
+  if (!derivationVersion) {
+    const config = getConfig();
+    if (config.network === 'mainnet') {
+      derivationVersion = hsm.enums.VERSION_OPTIONS.BIP32_HTR_MAIN_NET;
+    } else {
+      derivationVersion = hsm.enums.VERSION_OPTIONS.BIP32_HTR_TEST_NET;
+    }
+  }
+
+  // Derivation 1: Bip Code
+  await hsmConnection.blockchain.createBip32ChildKeyDerivation(
+    derivationVersion, // Version
+    hsm.enums.BCHAIN_SECURE_BIP32_INDEX.BASE + 44, // Derivation index
+    false, // Exportable
+    true, // Temporary
+    hsmKeyName, // Parent key name
+    childKeyNames.HTR_CKD_BIP_KEYNAME, // Child key name
+  );
+
+  // Derivation 2: Coin
+  await hsmConnection.blockchain.createBip32ChildKeyDerivation(
+    derivationVersion,
+    hsm.enums.BCHAIN_SECURE_BIP32_INDEX.BASE + 280,
+    false,
+    true,
+    childKeyNames.HTR_CKD_BIP_KEYNAME,
+    childKeyNames.HTR_CKD_COIN_KEYNAME,
+  );
+
+  // Derivation 3: Account
+  await hsmConnection.blockchain.createBip32ChildKeyDerivation(
+    derivationVersion,
+    hsm.enums.BCHAIN_SECURE_BIP32_INDEX.BASE,
+    false,
+    true,
+    childKeyNames.HTR_CKD_COIN_KEYNAME,
+    childKeyNames.HTR_CKD_ACCOUNT_KEYNAME,
+  );
+
+  if (options.isReadOnlyWallet) {
+    return {
+      success: true,
+      htrKeyName: childKeyNames.HTR_CKD_ACCOUNT_KEYNAME,
+    };
+  }
+
+  // Derivation 4: Change
+  await hsmConnection.blockchain.createBip32ChildKeyDerivation(
+    derivationVersion,
+    0,
+    false,
+    true,
+    childKeyNames.HTR_CKD_ACCOUNT_KEYNAME,
+    childKeyNames.HTR_CKD_CHANGE_KEYNAME,
+  );
+
+  return {
+    success: true,
+    htrKeyName: childKeyNames.HTR_CKD_CHANGE_KEYNAME,
+  };
+}
+
+/**
+ * Derivates an HTR wallet xPub string from a given HSM xPriv key
+ * @param {Object} hsmConnection
+ * @param {string} hsmKeyName
+ * @param {Object} [options]
```

Comment:
What are the options here?

Reply:
As discussed on the comment above, this parameter was removed.